### PR TITLE
Use head version of all supported, released MRI versions

### PR DESCRIPTION
**Pull request by calebthompson**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,11 +7,10 @@ script: bundle exec rake travis
 matrix:
   fast_finish: true
   include:
-    - rvm: 1.9.3
```
Comment (plribeiro3000) [inline .travis.yml]: Could you not remove `1.9.3` yet?

File: .travis.yml [hunk 2]
```diff
@@ -7,11 +7,10 @@ script: bundle exec rake travis
...
-    - rvm: 2.0.0
-    - rvm: 2.1.0
-    - rvm: 2.1.5
-    - rvm: 2.1.5
+    - rvm: 2.0
+    - rvm: 2.1
+    - rvm: 2.2
+    - rvm: 2.2
```
Comment (plribeiro3000) [inline .travis.yml]: You have `2.2` twice. I know it was wrong before but perhaps you can fix it as well?

Comment (calebthompson) [inline .travis.yml]: I believe this second one is to run on 2.2 with the head versions of gems, whereas the previous 2.2 uses the released versions.

File: .travis.yml (hunk 1 above)
Comment (calebthompson) [inline .travis.yml]: Are you sure? https://www.ruby-lang.org/en/news/2015/02/23/support-for-ruby-1-9-3-has-ended/

File: fog.gemspec [hunk 3]
```diff
@@ -45,7 +45,7 @@ Gem::Specification.new do |s|
   s.extra_rdoc_files = %w[README.md]
 
   ## Ruby version
-  s.required_ruby_version = ">= 1.9.3"
+  s.required_ruby_version = ">= 2"
```
Comment (plribeiro3000) [inline fog.gemspec]: We didn't get to a decision about dropping `1.9.3` so perhaps we need to know if for sure first.

File: .travis.yml (hunk 2 above)
Comment (plribeiro3000) [inline .travis.yml]: But it does not make sense, does it?

File: .travis.yml (hunk 1 above)
Comment (plribeiro3000) [inline .travis.yml]: We didn't decide about `1.9.3` yet. Perhaps the end of support might change this scenario.

@tokengeek @geemus Thoughts?

File: .travis.yml (hunk 2 above)
Comment (calebthompson) [inline .travis.yml]: I think it does. We're doing a separate build of our changes in `fog` against released versions of `fog-core` and `fog-json` and also against the master branches of those two gems.

Comment (plribeiro3000) [inline .travis.yml]: But you don't need to list it twice to play with the matrix, do you?

Comment (calebthompson) [inline .travis.yml]: I'm not entirely sure. It appears that master has duplicate entries as well. https://github.com/fog/fog/blob/master/.travis.yml#L22-L25

This is how the Travis docs recommend, but we end up with several more runs (each version against edge), which could be good:

```yml
rvm:
  - 2.0
  - 2.1
  - 2.2
  - jruby-19mode
  - jruby-head

gemfile:
  - Gemfile
  - gemfiles/Gemfile-edge

matrix:
  fast_finish: true

  allow_failures:
    - rvm: jruby-head
```

Comment (calebthompson) [inline .travis.yml]: http://docs.travis-ci.com/user/build-configuration/

Comment (calebthompson) [inline .travis.yml]: Do you want that instead?

Comment (plribeiro3000) [inline .travis.yml]: Yeah. I think that the duplication wrong and is not needed.

Comment (plribeiro3000) [inline .travis.yml]: Awesome. This way it is more clear and should be the same as before. :+1:

File: fog.gemspec (hunk 3 above)
Comment (geemus) [inline fog.gemspec]: I think we should probably leave this as 1.9.3 for now, though dropping it also in time would be good.

**Comment by calebthompson** [comment]
1.9.3 support re-added.

**Comment by geemus** [comment]
Thanks!

**Comment by plribeiro3000** [comment]
@calebthompson Could you add `2.2` as an allowed failure for now?

File: lib/fog.rb [hunk 4]
```diff
@@ -23,6 +23,7 @@
 require 'fog/aws'
 require 'fog/bluebox'
 require 'fog/brightbox'
+require 'fog/cloudfoundry'
```
Comment (tokengeek) [inline lib/fog.rb]: This needs removing since it's unrelated to the PR and Travis can't load it.

**Comment by tokengeek** [comment]
Re: 1.9.3 support - this is still in place for the foreseeable future because lots of linux distros in production seem to have 1.9.3 as the default.

Obviously this branch/work is for a major change but when we discussed it (https://github.com/fog/fog/pull/3348#issuecomment-67451436) that was raised as a issue.

I need to catch @geemus and @krames (maybe @plribeiro3000 ) and finalise a minimal "roadmap" (not scattered across dozens of tickets) with a few "major" releases over the next few months. Then 1.9.3 can go with 3.0 or 4.0. 2.0 being a minimal drop 1.8.7 release.

Unfortunately fog 2.0 has become the branding for the "next generation" of fog which was a mistake on my part.

**Comment by plribeiro3000** [comment]
@calebthompson Could you take care of @tokengeek comment?
I think it is the last point to get this merged. :+1:

**Comment by calebthompson** [comment]
Done.

**Comment by plribeiro3000** [comment]
Thanks @calebthompson!

**Comment by geemus** [comment]
Thanks!

**Pull request merged**